arcgis-ios-sdk-samples/Shared resources/Libraries/DemoTouchView/DemoTouchView.swift

```diff
@@ -108,58 +108,27 @@ extension UIApplication {
         
         if let touches = event.allTouches {
             
-            var began: Set<UITouch>?
-            var moved: Set<UITouch>?
-            var cancelled: Set<UITouch>?
-            var ended: Set<UITouch>?
-            
-            for touch in touches {
```

Comment:
The `for` loop was actually more efficient than four separate calls to `filter`. Not a big deal because the size of `allTouches` is going to be fairly small, but something to keep in mind. One way of simplifying this method while maintaining efficiency is by extending `UIEvent`:

```swift
extension UIEvent {
    var touchesByPhase: [UITouch.Phase: Set<UITouch>] {
        return allTouches?.reduce(into: [UITouch.Phase: Set<UITouch>](), { (result, touch) in
            result[touch.phase, default: []].insert(touch)
        }) ?? [:]
    }
}
```